### PR TITLE
Refactor string split helper method into core attributes.

### PR DESCRIPTION
**Pull request by DragonMinded**
## Summary

We have a helper in one of our codemods that should really be universally available. So, move the relevant bits into libcst core, remove them from the codemod, update the codemod to use the new behavior and update the tests accordingly.

## Test Plan

tox, pyre, new tests

### CODE REVIEW

File: libcst/_nodes/expression.py [hunk 1]
```diff
@@ -517,8 +517,23 @@ class BaseString(BaseExpression, ABC):
...
+        Returns the string's prefix, if any exists.
+
+        See `String and Bytes literals
+        <https://docs.python.org/3.7/reference/lexical_analysis.html#string-and-bytes-literals>`_
+        for more information.
+        """
+        ...
+
+    @property
+    def quote(self) -> str:
```
Comment (jimmylai) [inline libcst/_nodes/expression.py]: Can we use Literal to be specific?
```suggestion
    def quote(self) -> Literal['"', "'", ...]:
```

Comment (DragonMinded) [reply libcst/_nodes/expression.py]: I think that will fix typing errors when sending the result back to a new string/f-string construction too.

File: libcst/_nodes/expression.py [hunk 2]
```diff
@@ -516,9 +516,27 @@ class BaseString(BaseExpression, ABC):
...
 
 
+StringQuoteLiteral = Literal['"', "'", '"""', "'''"]
+
+
 class _BasePrefixedString(BaseString, ABC):
-    @abstractmethod
-    def _get_prefix(self) -> str:
+    @property
+    def prefix(self) -> str:
```
Comment (jimmylai) [inline libcst/_nodes/expression.py]: Can we also use Literal to type prefix? It's also a limited set of strings.

Comment (DragonMinded) [reply libcst/_nodes/expression.py]: Literal makes less sense here, as prefix validity is version-dependent.

**Pull request merged**